### PR TITLE
Fix SPM support

### PR DESCRIPTION
**Pull request by axelpart**
Hello Nickolay,

This pull request updates the package implementation to a newer Swift tools version. Would be nice if you can merge it.
Thank you in advance.

Alexandros

### CODE REVIEW

File: Sources/UI/UITableView+Diff.swift [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import UIKit
+import OptiDiff
```
Comment (tarbayev) [inline Sources/UI/UITableView+Diff.swift]: Let's make the linter happy:
```suggestion
import OptiDiff
import UIKit
```

Comment (axelpart) [reply Sources/UI/UITableView+Diff.swift]: I see you did the change already thank you. 💪

**Comment by axelpart** [comment]
I see that it fails now in test action because of Swift tools mismatch. I can downgrade from 5.8 to 5.7.1 what do you think?

**Comment by tarbayev** [comment]
@axelpart I updated the CI to support 5.8, now is should work.

File: Sources/UI/UITableView+Diff.swift [hunk 2]
```diff
@@ -1,3 +1,4 @@
...
```
Comment (tarbayev) [inline Sources/UI/UITableView+Diff.swift]: This causes issues in CocoaPods, let's try `canImport`:
```suggestion
#if canImport(OptiDiff)
  import OptiDiff
#endif
```

Comment (tarbayev) [reply Sources/UI/UITableView+Diff.swift]: Didn't work. Need to find another way of organizing the modules.

File: Sources/UI/UITableView+Diff.swift [hunk 3]
```diff
@@ -1,5 +1,5 @@
-#if canImport(OptiDiff)
-  import OptiDiff
+#if SWIFT_PACKAGE
```
Comment (tarbayev) [inline Sources/UI/UITableView+Diff.swift:1]: Nice! 
Please run SwiftFormat for your changes:
```
./swiftformat .
```

Comment (axelpart) [reply Sources/UI/UITableView+Diff.swift:1]: Thanks. Should be fine now.

**Review by tarbayev — APPROVED** [review]

**Comment by tarbayev** [comment]
@axelpart Thanks for your contribution!

**Comment by axelpart** [comment]
> @axelpart Thanks for your contribution!

Thank you too Nickolay for merging this one.

**Pull request merged**